control/cli.py

```diff
@@ -156,298 +207,1590 @@ def connect(self, host, port, client_key, client_cert, server_cert):
+                    if args.nsid:
+                        prefix = f"Namespace {args.nsid} in"
+                    elif args.uuid:
+                        prefix = f"Namespace with UUID {args.uuid} in"
+                    else:
+                        prefix = "Namespaces in"
+                    out_func(f"{prefix} subsystem {args.subsystem}:\n{namespaces_out}")
+                else:
+                    if args.nsid:
+                        out_func(f"No namespace {args.nsid} in subsystem {args.subsystem}")
+                    elif args.uuid:
+                        out_func(f"No namespace with UUID {args.uuid} in subsystem {args.subsystem}")
+                    else:
+                        out_func(f"No namespaces in subsystem {args.subsystem}")
+            else:
+                err_func(f"{namespaces_info.error_message}")
+        elif args.format == "json" or args.format == "yaml":
+            ret_str = json_format.MessageToJson(
+                        namespaces_info,
+                        indent=4,
+                        including_default_value_fields=True,
+                        preserving_proto_field_name=True)
+            if args.format == "json":
+                out_func(f"{ret_str}")
+            elif args.format == "yaml":
+                obj = json.loads(ret_str)
+                out_func(yaml.dump(obj))
+        elif args.format == "python":
+            return namespaces_info
+        else:
+            assert False
+
+        return namespaces_info.status
+
+    def ns_get_io_stats(self, args):
+        """Get namespace IO statistics."""
+
+        out_func, err_func = self.get_output_functions(args)
+        if not args.nsid and not args.uuid:
+            self.cli.parser.error("At least one of --nsid or --uuid arguments is mandatory for get_io_stats command")
+        if args.nsid and args.nsid < 0:
+            self.cli.parser.error("nsid value must be positive")
+        if args.size != None:
+            self.cli.parser.error("--size argument is not allowed for get_io_stats command")
+        if args.block_size != None:
+            self.cli.parser.error("--block-size argument is not allowed for get_io_stats command")
+        if args.rbd_pool != None:
+            self.cli.parser.error("--rbd-pool argument is not allowed for get_io_stats command")
+        if args.rbd_image != None:
+            self.cli.parser.error("--rbd-image argument is not allowed for get_io_stats command")
+        if args.load_balancing_group != None:
+            self.cli.parser.error("--load-balancing-group argument is not allowed for get_io_stats command")
+        if args.rw_ios_per_second != None:
+            self.cli.parser.error("--rw-ios-per-second argument is not allowed for get_io_stats command")
+        if args.rw_megabytes_per_second != None:
+            self.cli.parser.error("--rw-megabytes-per-second argument is not allowed for get_io_stats command")
+        if args.r_megabytes_per_second != None:
```

Comment:
First of all, I don't like too much the way the `@cli.cmd` decorator works, as it obfuscates the actual parameters accepted by a method, and complicates debugging.

I've checked the most popular Python cli parsing tools, and looks like the closest to this would be [`click`](https://click.palletsprojects.com/en/8.1.x/quickstart/#nesting-commands):
```python
@click.group
def ns():
  pass

@ns.command('add')
@click.option('--rbd-pool', default="rbd", help="RBD pool name")
@click.argument('subsystem', help="Subsystem NQN")
def ns_add(rbd_pool, subsystem):
  ...

@ns.command('list')
@click.option('--rbd-pool', default="rbd", help="RBD pool name")
def ns_list(rbd_pool)
  ...
```